src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -193,23 +193,39 @@ public IList<IProvider> Children
     [Serializable]
     public class ProviderCategoryConfiguration
     {
-        public string Name { get; set; }
+        public ProviderCategoryConfiguration()
+        {
+            Providers = new Dictionary<string, IProviderConfiguration>();
+        }
 
-        public IDictionary<string, IProviderConfiguration> Providers { get; set; }
+        internal ProviderCategoryConfiguration(string name) : this()
```

Comment:
I don't agree with you, sorry. I actually doubt that you've _actually_ used static checkers before. With Code Contracts, the first warning you will get on such a code is about _unnecessary null check_, since prover knows that variable was already checked higher in the stack. 

What is the point of writing the following code:

```cs
public void Foo(string s)
{
    if (s != null)
       Bar(s);
}

private void Bar(string s)
{
    if (s != null)
       Console.WriteLine(s);
}
```